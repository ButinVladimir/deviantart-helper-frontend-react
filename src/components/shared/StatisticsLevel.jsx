import React from 'react';
import PropTypes from 'prop-types';
import Level from 'react-bulma-components/lib/components/level';
import Content from 'react-bulma-components/lib/components/content';
import Heading from 'react-bulma-components/lib/components/heading';

export default function StatisticsLevel({
  titleValuePairs,
}) {
  const items = titleValuePairs.map(pair => (
    <Level.Item key={pair[0] + pair[1]} className="has-text-centered">
      <Content>
        <div>
          <Heading size={4}>{pair[1]}</Heading>
        </div>
        <div>
          <Heading size={6} subtitle textTransform="uppercase">{pair[0]}</Heading>
        </div>
      </Content>
    </Level.Item>
  ));

  return (
    <Level breakpoint="tablet">
      {items}
    </Level>
  );
}

StatisticsLevel.propTypes = {
  titleValuePairs: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.any)).isRequired,
};
