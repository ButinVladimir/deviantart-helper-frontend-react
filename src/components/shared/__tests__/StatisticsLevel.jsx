import React from 'react';
import { shallow } from 'enzyme';
import StatisticsLevel from '../StatisticsLevel';

describe('StatisticsLevel', () => {
  it('can be rendered correctly', () => {
    const titleValuePairs = [
      ['Views', 1],
      ['Favourites', 2],
      ['Comments', 3],
      ['Downloads', 4],
    ];
    const wrapper = shallow(<StatisticsLevel titleValuePairs={titleValuePairs} />);

    expect(wrapper).toMatchSnapshot();
  });
});
