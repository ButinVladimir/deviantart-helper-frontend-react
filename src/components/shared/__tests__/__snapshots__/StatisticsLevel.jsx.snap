// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticsLevel can be rendered correctly 1`] = `
<Level
  breakpoint="tablet"
  className=""
  renderAs="div"
  style={Object {}}
>
  <LevelItem
    className="has-text-centered"
    key="Views1"
    renderAs="div"
    style={Object {}}
  >
    <Content
      renderAs="div"
      size={null}
      style={Object {}}
    >
      <div>
        <Heading
          className=""
          heading={false}
          renderAs="h1"
          size={4}
          spaced={false}
          subtitle={false}
          weight={null}
        >
          1
        </Heading>
      </div>
      <div>
        <Heading
          className=""
          heading={false}
          renderAs="h1"
          size={6}
          spaced={false}
          subtitle={true}
          textTransform="uppercase"
          weight={null}
        >
          Views
        </Heading>
      </div>
    </Content>
  </LevelItem>
  <LevelItem
    className="has-text-centered"
    key="Favourites2"
    renderAs="div"
    style={Object {}}
  >
    <Content
      renderAs="div"
      size={null}
      style={Object {}}
    >
      <div>
        <Heading
          className=""
          heading={false}
          renderAs="h1"
          size={4}
          spaced={false}
          subtitle={false}
          weight={null}
        >
          2
        </Heading>
      </div>
      <div>
        <Heading
          className=""
          heading={false}
          renderAs="h1"
          size={6}
          spaced={false}
          subtitle={true}
          textTransform="uppercase"
          weight={null}
        >
          Favourites
        </Heading>
      </div>
    </Content>
  </LevelItem>
  <LevelItem
    className="has-text-centered"
    key="Comments3"
    renderAs="div"
    style={Object {}}
  >
    <Content
      renderAs="div"
      size={null}
      style={Object {}}
    >
      <div>
        <Heading
          className=""
          heading={false}
          renderAs="h1"
          size={4}
          spaced={false}
          subtitle={false}
          weight={null}
        >
          3
        </Heading>
      </div>
      <div>
        <Heading
          className=""
          heading={false}
          renderAs="h1"
          size={6}
          spaced={false}
          subtitle={true}
          textTransform="uppercase"
          weight={null}
        >
          Comments
        </Heading>
      </div>
    </Content>
  </LevelItem>
  <LevelItem
    className="has-text-centered"
    key="Downloads4"
    renderAs="div"
    style={Object {}}
  >
    <Content
      renderAs="div"
      size={null}
      style={Object {}}
    >
      <div>
        <Heading
          className=""
          heading={false}
          renderAs="h1"
          size={4}
          spaced={false}
          subtitle={false}
          weight={null}
        >
          4
        </Heading>
      </div>
      <div>
        <Heading
          className=""
          heading={false}
          renderAs="h1"
          size={6}
          spaced={false}
          subtitle={true}
          textTransform="uppercase"
          weight={null}
        >
          Downloads
        </Heading>
      </div>
    </Content>
  </LevelItem>
</Level>
`;
